gitconfig/syntax_test_gitconfig.gitconfig

```diff
@@ -0,0 +1,186 @@
+#                     ^^ string.unquoted.value
+#                        ^^^^^^ string.unquoted.value
+
+    pushf = push --force-with-lease
+# <- meta.key-value-pair
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^^^ meta.key variable.language
+#        ^ -variable.language
+#         ^ punctuation.separator.key-value
+#          ^ -punctuation.separator.key-value
+#           ^^^^^^^^^^^^^^^^^^^^^^^ meta.value
+#           ^^^^ string.unquoted.value
+#                ^^^^^^^^^^^^^^^^^^ string.unquoted.value
+    sla = log --oneline --decorate --graph --all
+# <- meta.key-value-pair
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ meta.key variable.language
+#      ^ -variable.language
+#       ^ punctuation.separator.key-value
+#        ^ -punctuation.separator.key-value
+#         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.value
+#         ^^^ string.unquoted.value
+#             ^^^^^^^^^
+#                       ^^^^^^^^^^ string.unquoted.value
+#                                  ^^^^^^^ string.unquoted.value
+#                                          ^^^^^ string.unquoted.value
+
+# ESCAPES TEST
+    foo = "\\ \" \b \n \t"
+#^^^^^^^^^^^^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^^^^^^^^^^^^^^^ meta.value string.quoted.double
+#          ^^ constant.character.escape
+#             ^^ constant.character.escape
+#                ^^ constant.character.escape
+#                   ^^ constant.character.escape
+#                      ^^ constant.character.escape
+
+
+# BOOLEAN TEST
+    foo = true
+#^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^^^ meta.value constant.language
+    foo = false
+#^^^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^^^^ meta.value constant.language
+    foo = on
+#^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^ meta.value constant.language
+    foo = off
+#^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^^ meta.value constant.language
+    foo = no
+#^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^ meta.value constant.language
+    foo = yes
+#^^^^^^^^^^^^ meta.key-value-pair
+#   ^^^ variable.language
+#       ^ punctuation.separator.key-value
+#         ^^^ meta.value constant.language
+
+# STRAY BRACKET TEST
+[foo]]
+#    ^ punctuation.definition.section.end invalid.illegal.stray-bracket
+
+# STRINGS TEST
+    foo = "foo"
+#         ^^^^^ meta.value string.quoted.double
+#         ^ punctuation.definition.string.begin
+#             ^ punctuation.definition.string.end
+    foo = \"foo\"
+#         ^^^^^^^ meta.value string.quoted.double
```

Comment:
I don't think this should be a double quoted string. It's an unquoted string with escaped double quotes.

Reply:
I like this 👍. I'll update the PR shortly with the improved scopes.